overlord/snapstate/storehelpers.go

```diff
@@ -69,6 +70,28 @@ func userFromUserID(st *state.State, userIDs ...int) (*auth.UserState, error) {
 	return user, err
 }
 
+func refreshOptions(st *state.State, origOpts *store.RefreshOptions) (*store.RefreshOptions, error) {
+	var opts store.RefreshOptions
+
+	if origOpts != nil {
+		if origOpts.RequestSalt != "" {
+			// nothing to add
+			return origOpts, nil
+		}
+		opts = *origOpts
+	}
+
+	if err := st.Get("refresh-request-salt", &opts.RequestSalt); err != nil && err != state.ErrNoState {
+		return nil, fmt.Errorf("cannot obtain store request salt: %v", err)
+	}
+	if opts.RequestSalt == "" {
+		// no request seed yet, make one now
+		opts.RequestSalt = time.Now().Format(time.RFC3339Nano)
+		st.Set("refresh-request-salt", opts.RequestSalt)
```

Comment:
I'm not sure what the thread model is for adding the salt. But if we add salt I am not sure this is the best choice. The data is quite predicable. I would suggest to use strutil.MakeRandomString() instead with the size we want (sha512crypt in glibc uses 16 chars iirc).

Reply:
Added.